### PR TITLE
Adds Vigere Shamir for Shields - Heal when blocking

### PR DESCRIPTION
**Pull request by EpyonProjects**
Adds a Barimium Shamir called Vigere, where in addition to blocking damage when using a shield, the user heals 1.5 hearts.

It just reimplements Sweethearts.  The idea was @TheThanathor's, so it is attributed to him.

### CODE REVIEW

File: gm4_vigere_shamir/pack.mcmeta [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    "wiki_link": "https://wiki.gm4.co/wiki/Metallurgy/Vigere_Shamir",
+    "required_modules": [
+        "metallurgy"
+    ],
+    "recommended_modules": [],
+    "credits": {
+        "Creator": [
+            [
+                "Thanathor",
+                "https://twitter.com/The_Thanathor"
```
Comment (BluePsychoRanger) [inline gm4_vigere_shamir/pack.mcmeta:31]: Did Epyon not also help create this Shamir?

**Review by BluePsychoRanger — APPROVED** [review]
Looks good. You should add the pack.png and pack.svg tho

File: gm4_vigere_shamir/data/gm4_vigere_shamir/functions/heal.mcfunction [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+# at @s, player blocking with a Vigere Shield
+# runs from gm4_vigere_shamir:main
+
+execute store result score @s gm4_vs_data run attribute @s minecraft:generic.max_health get
+
+execute store result score remove_health gm4_vs_data run attribute @s minecraft:generic.max_health get
+scoreboard players operation remove_health gm4_vs_data -= @s gm4_health
+
+# lower player's max health to their current health
+execute if score remove_health gm4_vs_data matches 512.. run attribute @s minecraft:generic.max_health modifier add e0c18b32-4caf-4193-8899-74ad1cc0dba2 gm4_vs_remove_512 -512 add
```
Comment (Bloo-dev) [inline gm4_vigere_shamir/data/gm4_vigere_shamir/functions/heal.mcfunction:11]: Fake players should use `$`, so `$remove_health`.

File: gm4_vigere_shamir/data/gm4_vigere_shamir/functions/heal.mcfunction [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+execute if score remove_health gm4_vs_data matches 8.. run attribute @s minecraft:generic.max_health modifier add 1f624b8a-89a4-4951-b4ce-f21890adbf89 gm4_vs_remove_8 -8 add
+execute if score remove_health gm4_vs_data matches 8.. run scoreboard players remove remove_health gm4_vs_data 8
+execute if score remove_health gm4_vs_data matches 4.. run attribute @s minecraft:generic.max_health modifier add 52b90103-a8c1-40a1-b2dd-9e7feb978b52 gm4_vs_remove_4 -4 add
+execute if score remove_health gm4_vs_data matches 4.. run scoreboard players remove remove_health gm4_vs_data 4
+execute if score remove_health gm4_vs_data matches 2.. run attribute @s minecraft:generic.max_health modifier add c875bec0-f7e4-4479-a030-dc92eb39d29a gm4_vs_remove_2 -2 add
+execute if score remove_health gm4_vs_data matches 2.. run scoreboard players remove remove_health gm4_vs_data 2
+execute if score remove_health gm4_vs_data matches 1.. run attribute @s minecraft:generic.max_health modifier add 41b9fe4f-ce8e-4025-8215-d47f9321b853 gm4_vs_remove_1 -1 add
+execute if score remove_health gm4_vs_data matches 1.. run scoreboard players remove remove_health gm4_vs_data 1
```
Comment (Bloo-dev) [inline gm4_vigere_shamir/data/gm4_vigere_shamir/functions/heal.mcfunction:30]: You could reuse the objective `gm4_ml_data`, as metallurgy will always be installed alongside this.

**Review by Bloo-dev — CHANGES_REQUESTED** [review]
This looks good overall. There are three minor things that need to be resolved:

- [x] `$` on fake player
- [ ] Use `gm4_ml_data`
- [ ] implement Lue's player heads (see example a) below)


a) You'll need to make a minor change to the loot table `band.json` and add a line to `init.mcfunction`:
changes
Here are two example files:
- [init.mcfunction](https://github.com/Gamemode4Dev/GM4_Datapacks/blob/0b36bc7f7eef1b4987cf3025e0e3f4ad37ca8f7d/gm4_percurro_shamir/data/gm4_percurro_shamir/functions/init.mcfunction)
- [band.json](https://github.com/Gamemode4Dev/GM4_Datapacks/blob/0b36bc7f7eef1b4987cf3025e0e3f4ad37ca8f7d/gm4_percurro_shamir/data/gm4_percurro_shamir/loot_tables/band.json)
**Remember to** update the band type, color of the lore text, translation key and metal amounts to suit your shamir. You can cross check with some other shamirs like celaro, vecto, audere or helious to get some more examples.

File: gm4_vigere_shamir/data/gm4_metallurgy/tags/functions/update_skull_owner.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "values": [
+    "gm4_vigere_shamir:update_skull_owner"
+  ]
+}
```
Comment (Bloo-dev) [inline gm4_vigere_shamir/data/gm4_metallurgy/tags/functions/update_skull_owner.json:5]: You don't need this file as your Shamir was created - and will therefore be merged - after PR #593. This is only required to update data on items created before that PR.

File: gm4_vigere_shamir/data/gm4_vigere_shamir/functions/update_skull_owner.mcfunction [hunk 5]
```diff
@@ -0,0 +1 @@
+ execute if data storage gm4_metallurgy:temp/shamir {name:"vigere"} run data modify storage gm4_metallurgy:temp/shamir skull_owner set value "[Drop to Fix Item] gm4_vigere_shamir:band/v0"
```
Comment (Bloo-dev) [inline gm4_vigere_shamir/data/gm4_vigere_shamir/functions/update_skull_owner.mcfunction:1]: You don't need this file as your Shamir was created - and will therefore be merged - after PR #593. This is only required to update data on items created before that PR.

**Review by Bloo-dev — CHANGES_REQUESTED** [review]

**Review by Bloo-dev — APPROVED** [review]
All requested changes were implemented, looks good to me.

**Review by Bloo-dev — CHANGES_REQUESTED** [review]
After further testing in `1.17.1-rc2` we got 2 leftover hearts from this module. Further testing is required.

**Review by Bloo-dev — APPROVED** [review]
The leftover hearts seem to be a Minecraft bug. We'll keep an eye on this but this won't prevent this PR from being merged.

**Review by Bloo-dev — APPROVED** [review]
Tested in game and code-reviewed.

**Pull request merged**